AppDevAnnouncements/Classes/AnnouncementNetworking.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  AnnouncementNetworking.swift
+//  AppDevAnnouncements
+//
+//  Created by Gonzalo Gonzalez on 11/17/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import FutureNova
+
+public class AnnouncementNetworking {
+
+    static private var announcementPath: String?
+
+    static private var isConfigSetup: Bool = false
+
+    static private let networking: Networking = URLSession.shared.request
+    
+    static public func setupConfig(scheme: String, host: String, commonPath: String, announcementPath: String) {
+        if(scheme != "" && host != "" && commonPath != "" && announcementPath != "") {
+            Endpoint.setupEndpointConfig(scheme, host, commonPath)
+            self.announcementPath = announcementPath
+            self.isConfigSetup = true
+        } else {
+            self.isConfigSetup = false
+        }
+    }
+
+    static private func getAnnouncement() -> Future<Response> {
+        return networking(Endpoint.getAnnouncement(announcementPath!)).decode()
```

Comment:
can we unwrap safely instead of force unwrapping?

Reply:
I tried to unwrap safely, but I would need to return some Future<Response> given the announcementPath is nil. In fact, it is impossible for the announcementPath to be nil at this point since retrieveAnnouncement only calls getAnnouncement assuming the config is setup properly, which requires you pass in a non-optional, non-empty string. I'm going to make announcementPath non-optional with a default value of empty string to remove this ambiguity.

Reply:
we can make the return type also optional and unwrap the future when we call this function